src/server/main.cpp

```diff
@@ -45,166 +42,159 @@ int main()
-
-        g_ranchDirector->HandleEnterRanch(clientId, enterRanch);
-      });
-
-    ranchServer.RegisterCommandHandler(
-      alicia::CommandId::RanchSnapshot,
-      [](alicia::ClientId clientId, auto& buffer)
-      {
-        alicia::RanchCommandRanchSnapshot snapshot;
-        alicia::RanchCommandRanchSnapshot::Read(snapshot, buffer);
-
-        g_ranchDirector->HandleSnapshot(clientId, snapshot);
-      });
-
-    ranchServer.RegisterCommandHandler(
-      alicia::CommandId::RanchCmdAction,
-      [](alicia::ClientId clientId, auto& buffer)
-      {
-        alicia::RanchCommandRanchCmdAction cmdAction;
-        alicia::RanchCommandRanchCmdAction::Read(cmdAction, buffer);
-
-        g_ranchDirector->HandleCmdAction(clientId, cmdAction);
-      });
-
-    ranchServer.RegisterCommandHandler<alicia::RanchCommandRanchStuff>(
-      alicia::CommandId::RanchStuff,
-      [](alicia::ClientId clientId, auto& command)
-      {
-        g_ranchDirector->HandleRanchStuff(clientId, command);
-      });
-
-    ranchServer.Host("0.0.0.0", 10031);
-  });
+  std::jthread ranchThread(
+    []()
+    {
+      alicia::CommandServer ranchServer("Ranch");
+      g_ranchDirector = std::make_unique<alicia::RanchDirector>(ranchServer);
+
+      ranchServer.RegisterCommandHandler(
+        alicia::CommandId::RanchEnterRanch,
+        [](alicia::ClientId clientId, auto& buffer)
+        {
+          alicia::RanchCommandEnterRanch enterRanch;
+          alicia::RanchCommandEnterRanch::Read(enterRanch, buffer);
+
+          g_ranchDirector->HandleEnterRanch(clientId, enterRanch);
+        });
+
+      ranchServer.RegisterCommandHandler(
+        alicia::CommandId::RanchSnapshot,
+        [](alicia::ClientId clientId, auto& buffer)
+        {
+          alicia::RanchCommandRanchSnapshot snapshot;
+          alicia::RanchCommandRanchSnapshot::Read(snapshot, buffer);
+
+          g_ranchDirector->HandleSnapshot(clientId, snapshot);
+        });
+
+      ranchServer.RegisterCommandHandler(
+        alicia::CommandId::RanchCmdAction,
+        [](alicia::ClientId clientId, auto& buffer)
+        {
+          alicia::RanchCommandRanchCmdAction cmdAction;
+          alicia::RanchCommandRanchCmdAction::Read(cmdAction, buffer);
+
+          g_ranchDirector->HandleCmdAction(clientId, cmdAction);
+        });
+
+      ranchServer.RegisterCommandHandler<alicia::RanchCommandRanchStuff>(
+        alicia::CommandId::RanchStuff,
+        [](alicia::ClientId clientId, auto& command)
+        { g_ranchDirector->HandleRanchStuff(clientId, command); });
+
+      ranchServer.Host("0.0.0.0", 10031);
+    });
 
   // Messenger thread.
-  std::jthread messengerThread([]()
-  {
-    alicia::CommandServer messengerServer("Messenger");
-    // TODO: Messenger
-    messengerServer.Host("0.0.0.0", 10032);
-  });
+  std::jthread messengerThread(
+    []()
+    {
+      alicia::CommandServer messengerServer("Messenger");
+      // TODO: Messenger
+      messengerServer.Host("0.0.0.0", 10032);
```

Comment:
Since the messenger server is still TODO you don't have to do it, but it'd be useful to have a section for its ip and port in the settings too already